GAE/src/org/waterforpeople/mapping/app/web/rest/CartodbRestService.java

```diff
@@ -0,0 +1,305 @@
+                                    formId)));
+            return response;
+        } catch (IOException e) {
+            return response;
+        }
+    }
+
+    @RequestMapping(method = RequestMethod.GET, value = "surveys")
+    @ResponseBody
+    public Map<String, Object> listSurveys() {
+        Map<String, Object> response = new HashMap<>();
+        response.put("surveys", null);
+        try {
+            response.put("surveys", queryCartodb("SELECT * FROM survey"));
+            return response;
+        } catch (IOException e) {
+            return response;
+        }
+    }
+
+    @RequestMapping(method = RequestMethod.GET, value = "forms")
+    @ResponseBody
+    public Map<String, Object> getForms(@RequestParam("surveyId") Long surveyId) {
+        Map<String, Object> response = new HashMap<>();
+        response.put("forms", null);
+        try {
+            response.put("forms",
+                    queryCartodb(String.format("SELECT * FROM form WHERE survey_id=%d", surveyId)));
+            return response;
+        } catch (IOException e) {
+            return response;
+        }
+    }
+
+    @RequestMapping(method = RequestMethod.GET, value = "distinct")
+    @ResponseBody
+    public Map<String, Object> getDistinctValues(
+            @RequestParam("question_name") String questionName, @RequestParam("form_id") Long formId) {
+        Map<String, Object> response = new HashMap<>();
+        response.put("distinct_values", null);
+        try {
+            response.put("distinct_values",
+                    queryCartodb(String.format("SELECT DISTINCT %s FROM raw_data_%d", questionName,
+                            formId)));
+            return response;
+        } catch (IOException e) {
+            return response;
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    @RequestMapping(method = RequestMethod.GET, value = "named_maps")
+    @ResponseBody
+    public Map<String, Object> getNamedMaps() throws IOException {
+        HttpURLConnection connection = getConnection("GET", mapsApiURL());
+        return objectMapper.readValue(connection.getInputStream(), Map.class);
+    }
+
+    @RequestMapping(method = RequestMethod.POST, value = "named_maps")
+    @ResponseBody
+    public Map<String, Object> createNamedMaps(
+            @RequestBody NamedMapPayload payload)
+            throws IOException {
+
+        HttpURLConnection connection = getConnection("POST", mapsApiURL());
+        OutputStreamWriter writer = new OutputStreamWriter(connection.getOutputStream());
+        objectMapper.writeValue(writer, buildNamedMap(payload));
+        writer.close();
+
+        InputStream result = connection.getInputStream();
+
+        @SuppressWarnings("unchecked")
+        Map<String, Object> resultMap = objectMapper.readValue(result, Map.class);
+
+        return resultMap;
+    }
+
+    @RequestMapping(method = RequestMethod.POST, value = "update_map")
+    @ResponseBody
```

Comment:
Till now we have had the convention of `PUT` for updates.  I was wondering what was your reason to choose the `POST` here